task/monorail-server/src/bsp/sidecar_bc.rs

```diff
@@ -145,25 +146,15 @@ pub fn preinit() {
     while !seq.is_clock_config_loaded().unwrap_or(false) {
         sleep_for(10);
     }
-    // Wait for the front IO board to be configured (or for the board to
-    // be reported as missing).
-    loop {
-        let ready = seq.front_io_phy_ready();
-        match ready {
-            Ok(true) | Err(SeqError::NoFrontIOBoard) => break,
-            _ => sleep_for(10),
-        }
-    }
 }
 
 impl<'a, R: Vsc7448Rw> Bsp<'a, R> {
     /// Constructs and initializes a new BSP handle
     pub fn new(vsc7448: &'a Vsc7448<'a, R>) -> Result<Self, VscError> {
         let seq = Sequencer::from(SEQ.get_task_id());
-        let has_front_io = match seq.front_io_phy_ready() {
-            Ok(true) => true,
-            Err(SeqError::NoFrontIOBoard) => false,
-            _ => panic!("front IO board went away after preinit()"),
+        let has_front_io = match seq.front_io_board_present() {
```

Comment:
Since `front_io_board_present` is infallible on the `sidecar-seq` side, we could change the reply type in Idol to `Simple(bool)`.

Reply:
TIL! That looks much better indeed.